src/WorkBC.Web/Controllers/CareerProfilesController.cs

```diff
@@ -152,75 +150,58 @@ public async Task<ActionResult<SavedCareerProfile>> DeleteSavedCareerProfile(int
             return savedCareerProfile;
         }
 
-        // POST: api/CareerProfiles/Save/1234
+        // POST: api/career-profiles/Save/11101
         [HttpPost("save/{noc}")]
         public async Task<bool> SaveCareerProfile(string noc)
         {
             var nocList = noc.Split(new char[] { ',' }, StringSplitOptions.RemoveEmptyEntries);
             foreach (string nocCode in nocList)
             {
-                //find the career profile id based on the noc code
+                //Find the saved career profile id based on the noc code 2021 which is 5 digits long.
                 int careerProfileId = (
-                    from profile in _enterpriseContext.CareerProfiles
-                    join nocCodes in _enterpriseContext.Nocs on profile.NocId equals nocCodes.NocId
-                    where nocCodes.Nocyear == 2011 && nocCodes.Noccode == nocCode.PadLeft(4,'0')
-                    select profile.CareerProfileId
+                    from profile in _context.SavedCareerProfiles
```

Comment:
I think you need to also find by AspNetUserId, otherwise it won't save for my user if another user has already saved it.

Reply:
Agreed, rectified it.